### PR TITLE
Feature/reorganize recipe data

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======

- derive the composition dictionary from the structure of old recipes
- change `packing_priority` to `priority`
- Convert the rest of the header info in new recipes

Solution
========

- created composition dictionary and moved ingredients into `interior` 
- changed all `packing_priority` to `priority` through all files 
- convert the old recipe to v2 structure in _migrate_version 
- added tests 

with @meganrm 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by meganrm** [comment]
make sure to link the issues this PR addresses

File: cellpack/autopack/Environment.py [hunk 1]
```diff
@@ -1579,21 +1579,21 @@ def getSortedActiveIngredients(self, allIngredients):
         priorities1 = []
         ingr2 = []  # priority = 0 or none and will be assigned based on complexity
         priorities2 = []
-        ingr0 = []  # negative values will pack first in order of abs[packing_priority]
+        ingr0 = []  # negative values will pack first in order of abs[priority]
```
Comment (meganrm) [inline cellpack/autopack/Environment.py:1582]: only change in this file is packing_priority -> priority

File: cellpack/autopack/ingredient/Ingredient.py [hunk 2]
```diff
@@ -109,8 +109,8 @@ class Ingredient(Agent):
         - an optional pdb ID
```
Comment (meganrm) [inline cellpack/autopack/ingredient/Ingredient.py:109]: only change in this file is packing_priority -> priority

File: cellpack/autopack/ingredient/grow.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def __init__(
         count=0,
```
Comment (meganrm) [inline cellpack/autopack/ingredient/grow.py:64]: only change in this file is packing_priority -> priority

File: cellpack/autopack/loaders/recipe_loader.py [hunk 4]
```diff
@@ -204,14 +240,16 @@ def _read(self):
...
         if (
             "format_version" not in recipe_data
-            or recipe_data["format_version"] != self.latest_version
+            or recipe_data["format_version"] != self.current_version
         ):
-            format_version = (
+            input_format_version = (
                 recipe_data["format_version"]
                 if "format_version" in recipe_data
                 else "1.0"
```
Comment (meganrm) [inline cellpack/autopack/loaders/recipe_loader.py:248]: we should change this to be "1.0.0" to be consistent with our current version being "2.0.0". Or vice versa, i'm not sure we really need all 3 levels. but it should be consistent either way

File: cellpack/autopack/loaders/recipe_loader.py [hunk 5]
```diff
@@ -13,9 +13,11 @@
     v1_to_v2_name_map,
     unused_attributes_list,
     convert_to_partners_map,
+    attributes_move_to_composition,
 )
 
 encoder.FLOAT_REPR = lambda o: format(o, ".8g")
+CURRENT_VERSION = "2.0.0"
```
Comment (meganrm) [inline cellpack/autopack/loaders/recipe_loader.py:20]: Let's change this to "2.0", and keep all the format version just with 1 decimal place.

File: cellpack/autopack/loaders/recipe_loader.py [hunk 6]
```diff
@@ -180,22 +182,56 @@ def _migrate_ingredient(old_ingredient):
...
 
     @staticmethod
-    def _migrate_version(recipe, format_version):
-        if format_version == "1.0":
-            recipe["bounding_box"] = recipe["options"]["boundingBox"]
-            recipe["objects"] = RecipeLoader._get_v1_ingredients(recipe)
-        return recipe
+    def _migrate_version(recipe, current_version, format_version="1.0"):
+        new_recipe = {}
+        if format_version == "1.0" or format_version == "1.0.0":
```
Comment (meganrm) [inline cellpack/autopack/loaders/recipe_loader.py:225]: we won't need the "1.0.0" or statement here

**Review by meganrm — APPROVED** [review]

File: cellpack/autopack/Environment.py (hunk 1 above)
Comment (rugeli) [reply cellpack/autopack/Environment.py:1582]: thank you for the example, I will keep in mind to leave necessary comments to reviewers in large PRs.

**Pull request merged**